src/screens/TypeScreen.test.tsx

```diff
@@ -55,8 +53,68 @@ describe('TypeScreen', () => {
     expect(addStub).toHaveBeenCalledWith({
       account: 'My Account',
       issuer: 'Issuer A',
-      secret: 'mysecret',
+      secret: 'MYSECRET',
       uid: '11-22-33-44',
     })
   })
+
+  describe('when malformed secret has been input', () => {
+    describe('when it contains un-authorised digits', () => {
+      beforeEach(() => {
+        const { getByLabelText } = setup()
+
+        fireEvent.changeText(getByLabelText('Issuer'), 'Issuer A')
+        fireEvent.changeText(getByLabelText('Secret'), 'mysecret!1')
+        fireEvent.changeText(getByLabelText('Account'), 'My Account')
+      })
+
+      it('should show an error message', () => {
+        const { getByText } = screen
```

Comment:
folks let's try to be consistent with existing code too. either use screen or use the return value of setup() but avoid using both as it gets quite confusing